src/Containers/MovieListPage/MovieListPage.jsx

```diff
@@ -0,0 +1,93 @@
+import { useEffect, useState } from 'react';
+import { GenreSelect } from '../../Components/GenreSelect/GenreSelect';
+import { SearchForm } from '../../Components/SearchForm/SearchForm';
+import { sortOptions } from '../../constants';
+import { MovieDetails } from '../../Components/MovieDetails/MovieDetails';
+import { MovieTile } from '../../Components/MovieTile/MovieTile';
+import { SortControl } from '../../Components/SortControl/SortControl';
+import { Dialog } from '../../Components/Dialog/Dialog';
+import { MovieForm } from '../../Components/MovieForm/MovieForm';
+import styles from './MovieListPage.module.css';
+import { fetchMovies } from '../../requests/requests';
+
+export function MovieListPage() {
+  const [selectedGenre, setSelectedGenre] = useState('All');
+  const [searchQuery, setSearchQuery] = useState('');
+  const [selectedSortOption, setSelectedSortOption] = useState(sortOptions[0].value);
+  const [movies, setMovies] = useState([]);
+  const [selectedMovie, setSelectedMovie] = useState();
+  const genres = ['All', 'Comedy', 'Horror', 'Crime'];
+
+  const [isAddMovieDialogOpen, setIsAddMovieDialogOpen] = useState(false);
+
+  useEffect(() => {
+    const controller = new AbortController();
+
+    const fetchData = async () => {
+      const movies = await fetchMovies(searchQuery, selectedSortOption, selectedGenre, controller);
+      setMovies(movies);
+    };
+
+    fetchData();
+
+    return () => {
+      controller.abort();
+    };
+  }, [searchQuery, selectedSortOption, selectedGenre, setMovies]);
+
+  return (
+    <div>
+      <div className={styles.headerTile}>
+        {selectedMovie ? (
+          <>
+            <MovieDetails
+              imageUrl={selectedMovie.imageUrl}
+              movieName={selectedMovie.movieName}
+              releaseYear={selectedMovie.releaseYear}
+              rating={selectedMovie.rating}
+              duration={selectedMovie.duration}
+              description={selectedMovie.description}
+            />
+            <button className={styles.headerButton} onClick={() => setSelectedMovie(null)}>
+              Search
+            </button>
+          </>
+        ) : (
+          <>
+            <SearchForm initialSearchQuery="" onSearch={(value) => setSearchQuery(value)} />
```

Comment:
To have current search query in the input:

```suggestion
            <SearchForm initialSearchQuery={searchQuery} onSearch={(value) => setSearchQuery(value)} />
```